Build error for tests in "windows-msys2 / CC clang; GnuPG stable"

There is an error in "windows-msys2 / CC clang; GnuPG stable" that affected PR #1781.

```
-- Generating done
-- Build files have been written to: D:/a/_temp/msys64/home/runneradmin/local-builds/rnp-build
+ make_install
+ make -j2 install
[  1%] Building CXX object src/common/CMakeFiles/rnp-common.dir/str-utils.cpp.obj
[  1%] Building CXX object src/tests/googletest-build/googletest/CMakeFiles/gtest.dir/src/gtest-all.cc.obj
D:/a/rnp/rnp/src/common/str-utils.cpp:30:10: fatal error: 'cstddef' file not found
#include <cstddef>
         ^~~~~~~~~
1 error generated.
In file included from D:/a/_temp/msys64/home/runneradmin/googletest-download/googletest-src/googletest/src/gtest-all.cc:38:
D:/a/_temp/msys64/home/runneradmin/googletest-download/googletest-src/googletest/include/gtest/gtest.h:55:10: fatal error: 'cstddef' file not found
#include <cstddef>
         ^~~~~~~~~
make[2]: *** [src/common/CMakeFiles/rnp-common.dir/build.make:76: src/common/CMakeFiles/rnp-common.dir/str-utils.cpp.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:1166: src/common/CMakeFiles/rnp-common.dir/all] Error 2
```

This error is only present on this particular configuration for some reason. 

We need to fix this.

_Originally posted by @ronaldtse in https://github.com/rnpgp/rnp/issues/1781#issuecomment-1081272530_

--- a/.github/workflows/windows-msys2.yml
+++ b/.github/workflows/windows-msys2.yml
@@ -33,6 +33,7 @@ on:
 env:
   BUILD_MODE: normal
   GPG_VERSION: stable
+  GTEST_SOURCES: googletest-src
   CORES: 2
 
 jobs:
@@ -89,7 +90,7 @@ jobs:
         uses: actions/checkout@v3
         with:
           repository: google/googletest
-          path: googletest-src
+          path: ${{ env.GTEST_SOURCES }}
           fetch-depth: 0
           ref: release-1.11.0
         # This is workaround for the CMake or MSYS issue, failing to correctly checkout external project
@@ -97,8 +98,7 @@ jobs:
         #  fatal: ambiguous argument 'HEAD^commit': unknown revision or path not in the working tree.
       - name: tests
         shell: msys2 {0}
-        run: |
-          env GTEST_SOURCES=googletest-src ci/main.sh
+        run: ci/main.sh
       - shell: msys2 {0}
         run: |
           set -euxo pipefail

--- a/.github/workflows/windows-msys2.yml
+++ b/.github/workflows/windows-msys2.yml
@@ -1,6 +1,7 @@
 name: windows-msys2
 
 on:
+  workflow_dispatch:
   push:
     branches:
       - master
@@ -47,20 +48,19 @@ jobs:
         env:
           - CC: gcc
             CXX: g++
-            GTEST_SOURCES: /home/runneradmin/googletest-download/googletest-src
             RNP_INSTALL: /home/runneradmin/rnp-install
           - CC: clang
             CXX: clang++
-            GTEST_SOURCES: /home/runneradmin/googletest-download/googletest-src
             RNP_INSTALL: /home/runneradmin/rnp-install
     env:  ${{ matrix.env }}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 1
       - uses: msys2/setup-msys2@v2
         with:
-          path-type: inherit
+          path-type: minimal
+          update: true
       - name: Setup environment
         shell: msys2 {0}
 #  setup-env script below cannot correctly process msys2 environment since it is a mix of Windows and Linux
@@ -86,21 +86,19 @@ jobs:
 #          set -x
 #          ci/install_cacheable_dependencies.sh
       - name: download googletest
+        uses: actions/checkout@v3
+        with:
+          repository: google/googletest
+          path: googletest-src
+          fetch-depth: 0
+          ref: release-1.11.0
         # This is workaround for the CMake or MSYS issue, failing to correctly checkout external project
         #  Failed to get the hash for HEAD:
         #  fatal: ambiguous argument 'HEAD^commit': unknown revision or path not in the working tree.
-        shell: msys2 {0}
-        run: |
-          set -euxo pipefail
-          mkdir ${HOME}/googletest-download
-          pushd ${HOME}/googletest-download
-          git clone --no-checkout "https://github.com/google/googletest.git" "googletest-src"
-          cd googletest-src
-          git checkout c43f710 --
-          popd
       - name: tests
         shell: msys2 {0}
-        run: bash ci/main.sh
+        run: |
+          env GTEST_SOURCES=$(pwd)/googletest-src ci/main.sh
       - shell: msys2 {0}
         run: |
           set -euxo pipefail

--- a/.github/workflows/windows-msys2.yml
+++ b/.github/workflows/windows-msys2.yml
@@ -98,7 +98,7 @@ jobs:
       - name: tests
         shell: msys2 {0}
         run: |
-          env GTEST_SOURCES=$(pwd)/googletest-src ci/main.sh
+          env GTEST_SOURCES=googletest-src ci/main.sh
       - shell: msys2 {0}
         run: |
           set -euxo pipefail

--- a/ci/lib/install_functions.inc.sh
+++ b/ci/lib/install_functions.inc.sh
@@ -611,6 +611,7 @@ msys_install() {
     gnupg2
     make
     pkg-config
+    p7zip
     mingw64/mingw-w64-x86_64-cmake
     mingw64/mingw-w64-x86_64-python3
   )

